src/encoder/colortype.rs

```diff
@@ -0,0 +1,74 @@
+use crate::decoder::PhotometricInterpretation;
+
+/// Trait for different colortypes that can be encoded.
+pub trait ColorType {
+    /// The type of each sample of this colortype
+    type Inner: super::TiffValue;
+    /// The value of the tiff tag `PhotometricInterpretation`
+    const TIFF_VALUE: PhotometricInterpretation;
+    /// The value of the tiff tag `BitsPerSample`
+    fn bits_per_sample() -> Vec<u16>;
```

Comment:
I'm not sure why this is a function but the other is associated value is a constant. Could `const bits_per_sample: &'static [u16]` work as well?